src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1479,6 +1482,120 @@ class Simplex_tree {
+        }
+        if ((px <= -1) & (py >=  1)){
+          x = minval_ + (maxval_-minval_)*(px + 2); 
+          y = minval_ - (maxval_-minval_)*(py - 2);
+          if (x <= y){
+            new_dgm[2].push_back(std::make_pair(h, std::make_pair(x,y)));
+          }
+          else{  
+            new_dgm[3].push_back(std::make_pair(h, std::make_pair(x,y)));
+          }
+        }
+      }
+    }
+    return new_dgm;
+  }
+
+  /** \brief Extend filtration for computing extended persistence. 
+   * This function only uses the filtration values at the 0-dimensional simplices, 
+   * and computes the extended persistence diagram induced by the lower-star filtration 
+   * computed with these values. 
+   * \post Note that after calling this function, the filtration 
+   * values are actually modified. The function compute_extended_persistence_subdiagrams 
+   * retrieves the original values and separates the extended persistence diagram points 
+   * w.r.t. their types (Ord, Rel, Ext+, Ext-) and should always be called after 
+   * computing the persistent homology of the extended simplicial complex.
+   * \post Note that this code creates an extra vertex internally, so you should make sure that
+   * the Simplex tree does not contain a vertex with the largest Vertex_handle. 
+   */
+  void extend_filtration() {
+
+    // Compute maximum and minimum of filtration values
+    int maxvert = std::numeric_limits<int>::min();
+    this->minval_ = std::numeric_limits<double>::max();
+    this->maxval_ = std::numeric_limits<double>::min();
+    for (auto sh = root_.members().begin(); sh != root_.members().end(); ++sh){
+      double f = this->filtration(sh);
+      this->minval_ = std::min(this->minval_, f);
+      this->maxval_ = std::max(this->maxval_, f);
+      maxvert = std::max(sh->first, maxvert);
+    }
+    
+    GUDHI_CHECK(maxvert < std::numeric_limits<int>::max(), std::invalid_argument("Simplex_tree contains a vertex with the largest Vertex_handle"));
+    maxvert += 1;
+
+    Simplex_tree st_copy = *this;
+
+    // Add point for coning the simplicial complex
+    int count = this->num_simplices();
+    this->insert_simplex({maxvert}, -3);
+    count++;
+
+    // For each simplex
+    for (auto sh_copy : st_copy.complex_simplex_range()){
+
+      // Locate simplex
+      std::vector<Vertex_handle> vr;
+      for (auto vh : st_copy.simplex_vertex_range(sh_copy)){
+        vr.push_back(vh);
+      }
+      auto sh = this->find(vr);
+
+      // Create cone on simplex
+      vr.push_back(maxvert);
+      if (this->dimension(sh) == 0){
+        // Assign ascending value between -2 and -1 to vertex
+        double v = this->filtration(sh);
+        this->assign_filtration(sh, -2 + (v-this->minval_)/(this->maxval_-this->minval_));
+        // Assign descending value between 1 and 2 to cone on vertex
+        this->insert_simplex(vr, 2 - (v-this->minval_)/(this->maxval_-this->minval_));
```

Comment:
I would find it more readable if you stored (v-min)/(max-min) in a variable, so the reader doesn't have to check if this is exactly the same formula of if there is some subtlety (min vs max, sign...).

Reply:
Done.